notion-server/src/api/mod.rs

```diff
@@ -1,30 +1,80 @@
-use crate::ethereum::Handle;
+use crate::ethereum::{args, Handle, Result as EthResult};
+use crate::Notion;
 
-use rocket::{get, routes, State};
+use rocket::response::status;
+use rocket::{config, uri};
+use rocket::{get, post, routes, State};
 
-use snafu::Snafu;
+use rocket_contrib::json::Json;
+
+use snafu::{ResultExt, Snafu};
 
 /// Shorthand for result types returned from the API server.
 pub type Result<V, E = Error> = std::result::Result<V, E>;
 
 /// Errors arising from the API server.
 #[derive(Debug, Snafu)]
-pub enum Error {}
+pub enum Error {
+    Config { source: config::ConfigError },
+}
 
-pub fn run(handle: Handle) -> Result<()> {
-    // TODO: Get some configuration options from command line.
+pub fn run(notion: &Notion, handle: Handle) -> Result<()> {
+    let config = config::Config::build(config::Environment::Development)
+        .address(format!("{}", notion.bind.ip()))
```

Comment:
is this the canonical way to convert to string?

Reply:
Hm. There are several options. [`to_string`](https://doc.rust-lang.org/std/string/trait.ToString.html#tymethod.to_string) would work as well.

I'm so used to doing `format!("{:?}", foo)` that I didn't even think about it.

Would you like me to switch?